### PR TITLE
F003 calculate sale salary

### PR DESCRIPTION
**Pull request by Wos2610**
# **SUMARY**
Calculate salary for NhanVienKinhDoanh.

# **CHANGE**
Change in some classes:
- Class NhanVien: 
   - [x] Change to abstract class.
   - [x] Add new abstract computed property `realSalary`.
- Class NhanVienKinhDoanh: 
   - [x] Change method `toString()`
   - [x] Add new computed property `realSalary`.
- Class NhanVienLapTrinh: 
   - [x] Change method `toString()`
- Class QuanLy:
   - [x] Change method `inThongTin()`.

# **CHECKLIST**
- [x] Calculate salary for NhanVienKinhDoanh.
- [x] Edit output to clear information.
- [x] Reformat code with dart format.

# **RESULT**
image

### CODE REVIEW

File: NhanVien.dart [hunk 1]
```diff
@@ -6,9 +6,14 @@ class NhanVien{
...
   String diaChi;
   double luong;
+  double realSalary = 0;
   NhanVien({required this.hoTen, required this.ngaySinh, required this.soDT, required this.diaChi, required this.luong});
   @override
   String toString() {
     return '$maNV\t$hoTen\t$ngaySinh\t$soDT\t$diaChi\t$luong';
   }
+
+  void calSalary(){
```
Comment (Tris1702) [inline NhanVien.dart]: rename to `caculateSalary` for more meaningful

Comment (VuNA2k2) [inline NhanVien.dart]: what about using abstract class or interface in this case?

File: QuanLy.dart [hunk 2]
```diff
@@ -5,7 +5,9 @@ class QuanLy{
     List<NhanVien> dsNV = [];
     QuanLy(this.dsNV);
     void inThongTin(){
```
Comment (Tris1702) [inline QuanLy.dart]: Please follow code convention
Suggestion: from `){` to `) {`

File: QuanLy.dart [hunk 3]
```diff
@@ -5,7 +5,9 @@ class QuanLy{
...
+        print("maNV\thoTen\t\t\tngaySinh\tsoDT\t\tdiaChi\tluongCung\tdoanhSo/level\trealSalary");
         for(var nv in dsNV){
```
Comment (Tris1702) [inline QuanLy.dart]: Coding convention
Suggestion: from `){ `to `) {`

File: NhanVienKinhDoanh.dart [hunk 4]
```diff
@@ -12,6 +12,12 @@ class NhanVienKinhDoanh extends NhanVien {
     }
   @override
   String toString() {
-    return super.toString() + '\t$doanhSo';
+    return super.toString() + '\t\t$doanhSo\t\t\t\t$realSalary';
   }
+  @override
+  void calSalary(){
```
Comment (Tris1702) [inline NhanVienKinhDoanh.dart]: Code convention
Suggestion: from ){ to ) {

File: NhanVien.dart (hunk 1 above)
Comment (Tris1702) [reply NhanVien.dart]: code convention

File: NhanVienKinhDoanh.dart (hunk 4 above)
Comment (VuNA2k2) [inline NhanVienKinhDoanh.dart]: You should use return type instead of void.

**Review by Tris1702 — CHANGES_REQUESTED** [review]
When writing your code, please follow code convention!
Thanks, happy coding!!!

File: QuanLy.dart [hunk 5]
```diff
@@ -5,7 +5,9 @@ class QuanLy{
...
+            nv.calSalary();
```
Comment (VuNA2k2) [inline QuanLy.dart]: If you use a return type, you print the salary directly.

File: NhanVien.dart (hunk 1 above)
Comment (Wos2610) [reply NhanVien.dart]: I changed class NhanVien to abstract class.

File: NhanVienKinhDoanh.dart (hunk 4 above)
Comment (Wos2610) [reply NhanVienKinhDoanh.dart]: I changed to return double type.

File: NhanVien.dart (hunk 1 above)
Comment (Wos2610) [reply NhanVien.dart]: I reformatted code with dart format.

File: QuanLy.dart (hunk 2 above)
Comment (Wos2610) [reply QuanLy.dart]: I reformatted code with dart format.

File: QuanLy.dart (hunk 3 above)
Comment (Wos2610) [reply QuanLy.dart]: I reformatted code with dart format.

File: NhanVienKinhDoanh.dart (hunk 4 above)
Comment (Wos2610) [reply NhanVienKinhDoanh.dart]: I reformatted code with dart format.

File: NhanVien.dart [hunk 6]
```diff
@@ -1,14 +1,22 @@
...
+abstract class NhanVien {
   static int maNVCount = 1;
   String maNV = '${maNVCount++}';
   String hoTen;
   String ngaySinh;
   String soDT;
   String diaChi;
   double luong;
-  NhanVien({required this.hoTen, required this.ngaySinh, required this.soDT, required this.diaChi, required this.luong});
+  double realSalary = 0;
```
Comment (Tris1702) [inline NhanVien.dart]: This property is not necessary anymore

Comment (Tris1702) [reply NhanVien.dart]: change to get property

File: QuanLy.dart [hunk 7]
```diff
@@ -1,14 +1,15 @@
...
-            print(nv);
-        }
+class QuanLy {
+  List<NhanVien> dsNV = [];
+  QuanLy(this.dsNV);
+  void inThongTin() {
+    print(
+        "maNV\thoTen\t\t\tngaySinh\tsoDT\t\tdiaChi\tluongCung\tdoanhSo/level\trealSalary");
+    for (var nv in dsNV) {
+      nv.realSalary = nv.calculateSalary();
```
Comment (VuNA2k2) [inline QuanLy.dart]: Why do you need to set the value?

File: QuanLy.dart (hunk 5 above)
Comment (Wos2610) [reply QuanLy.dart]: I changed to print the salary by print `calculateSalary()`.

File: QuanLy.dart (hunk 7 above)
Comment (Wos2610) [reply QuanLy.dart]: I removed it.

File: NhanVien.dart (hunk 6 above)
Comment (Wos2610) [reply NhanVien.dart]: I removed `realSalary` and print salary by print `calculateSalary()`.

File: NhanVienKinhDoanh.dart [hunk 8]
```diff
@@ -1,17 +1,31 @@
...
+
   @override
   String toString() {
-    return super.toString() + '\t$doanhSo';
+    return super.toString() + '\t\t$doanhSo\t\t\t\t$realSalary';
   }
-}
\ No newline at end of file
+
+  double get realSalary => luong + doanhSo * 0.1;
```
Comment (Tris1702) [inline NhanVienKinhDoanh.dart:31]: missing `@override` annotation

File: NhanVienLapTrinh.dart [hunk 9]
```diff
@@ -1,18 +1,31 @@
...
+
   @override
   String toString() {
-    return super.toString() + '\t$level';
+    return super.toString() + '\t$level\t';
   }
 
-}
\ No newline at end of file
+  double get realSalary => 0;
```
Comment (Tris1702) [inline NhanVienLapTrinh.dart:31]: missing `@override` annotation

**Review by Tris1702 — CHANGES_REQUESTED** [review]

File: NhanVienKinhDoanh.dart (hunk 8 above)
Comment (Wos2610) [reply NhanVienKinhDoanh.dart:31]: done ạ

File: NhanVienLapTrinh.dart (hunk 9 above)
Comment (Wos2610) [reply NhanVienLapTrinh.dart:31]: done ạ

**Review by Tris1702 — APPROVED** [review]

**Review by VuNA2k2 — APPROVED** [review]

**Review by mdunggggg — APPROVED** [review]

**Pull request merged**